internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -142,7 +142,8 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 	}
 
 	if strings.HasPrefix(release.Commit(), newHash) {
-		return nil, ErrSameVersion
+		u.log.Error("Upgrade failed: upgrade did not occur because its the same version")
```

Comment:
Is this really an error and did it really fail? I would prefer to see:

```suggestion
		u.log.Warn("Upgrade action skipped: upgrade did not occur because its the same version")
```